### PR TITLE
Treasury updates

### PR DESCRIPTION
**Pull request by artur-abliazimov**

### CODE REVIEW

File: pool/src/lib.rs [hunk 1]
```diff
@@ -732,45 +747,49 @@ impl LendingPool {
...
 
         event::repay(env, who.clone(), asset.clone(), amount);
 
-        let remaning_amount = amount
-            .checked_sub(payback_amount)
-            .ok_or(Error::MathOverflowError)?;
-
-        let remaning_amount = if remaning_amount > 0 {
-            remaning_amount
+        let remaning_amount = if amount != i128::MAX && amount > borrower_actual_debt {
```
Comment (mn13) [inline pool/src/lib.rs:784]: it's not clear that i128::MAX works like `repay_only` flag, may be needs some rework

Comment (mn13) [reply pool/src/lib.rs:784]: repay_only can be removed, logic with i128::MAX must be described in method's doc

File: pool/src/lib.rs [hunk 2]
```diff
@@ -732,45 +740,49 @@ impl LendingPool {
...
-                .recip_mul_int(payback_amount)
-                .ok_or(Error::MathOverflowError)?
-        };
+        let borrower_debt_to_burn = debt_coeff
+            .recip_mul_int(borrower_payback_amount)
+            .ok_or(Error::MathOverflowError)?;
+        let s_token_transfer = collat_coeff
+            .mul_int(borrower_debt_to_burn)
+            .ok_or(Error::MathOverflowError)?;
+        let treasury_amount = borrower_payback_amount
```
Comment (rudewalt) [inline pool/src/lib.rs:767]:
```suggestion
        let treasury_part = borrower_payback_amount
```

File: pool/src/lib.rs [hunk 3]
```diff
@@ -732,45 +740,49 @@ impl LendingPool {
...
-            asset_debt
-        } else {
-            debt_coeff
-                .recip_mul_int(payback_amount)
-                .ok_or(Error::MathOverflowError)?
-        };
+        let borrower_debt_to_burn = debt_coeff
+            .recip_mul_int(borrower_payback_amount)
+            .ok_or(Error::MathOverflowError)?;
+        let s_token_transfer = collat_coeff
```
Comment (rudewalt) [inline pool/src/lib.rs:764]:
```suggestion
        let lenders_part = collat_coeff
```

**Review by mn13 — APPROVED** [review]

**Pull request merged**